azure-quantum/azure/quantum/job/base_job.py

```diff
@@ -0,0 +1,240 @@
+    input_data_format = None
+    output_data_format = None
+    provider_id = None
+
+    """
+    Base job class with methods to create a job from raw blob data,
+    upload blob data and download results.
+    """
+    @staticmethod
+    def create_job_id() -> str:
+        """Create a unique id for a new job."""
+        return str(uuid.uuid1())
+
+    @property
+    def container_name(self):
+        return f"job-{self.id}"
+
+    @classmethod
+    def from_input_data(
+        cls,
+        workspace: "Workspace",
+        name: str,
+        target: str,
+        input_data: bytes,
+        input_data_name: str,
+        encoding: str = "",
+        input_data_format: str = None,
+        output_data_format: str = None,
+        provider_id: str = None,
+        input_params: Dict[str, Any] = None
+    ) -> "BaseJob":
+        """Create a new Azure Quantum job based on a raw input_data payload.
+
+        :param workspace: Azure Quantum workspace to submit the input_data to
+        :type workspace: "Workspace"
+        :param name: Name of the job
+        :type name: str
+        :param target: Azure Quantum target
+        :type target: str
+        :param input_data: Raw input data to submit
+        :type input_data: bytes
+        :param input_data_name: input name
+        :type input_data_name: str
+        :param encoding: input_data encoding, e.g. "gzip", defaults to empty string
+        :type encoding: str
+        :param input_data_format: Input data format, defaults to None
+        :type input_data_format: str, optional
+        :param output_data_format: Output data format, defaults to None
+        :type output_data_format: str, optional
+        :param provider_id: Provider ID, defaults to None
+        :type provider_id: str, optional
+        :param input_params: Input parameters, defaults to None
+        :type input_params: Dict[str, Any], optional
+        :param input_params: Input params for job
+        :type input_params: Dict[str, Any]
+        :return: Azure Quantum Job
+        :rtype: Job
+        """
+        job_id = cls.create_job_id()
+        container_name = f"job-{job_id}"
+        container_uri = cls.create_container(
+            workspace=workspace,
+            container_name=container_name
+        )
+        logger.debug(f"Container URI: {container_uri}")
+        input_data_uri = cls.upload_input_data(
+            data=input_data,
+            input_data_name=input_data_name,
+            encoding=encoding
+        )
+
+        return cls.from_uri(
+            workspace=workspace,
+            job_id=job_id,
+            target=target,
+            input_data_uri=input_data_uri,
+            container_uri=container_uri,
+            name=name,
+            input_data_format=input_data_format,
+            output_data_format=output_data_format,
+            provider_id=provider_id,
+            input_params=input_params
+        )
+
+    @classmethod
+    def from_uri(
+        cls,
```

Comment:
it's not clear that `uri` here refers to an existing storage ur, so maybe name this `from_storage_uri`)?